sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/Program.cs

```diff
@@ -73,75 +52,33 @@ public static async Task Main(string[] args)
 
             var publishTelemetrySamples = new PublishTelemetrySamples();
             await publishTelemetrySamples.RunSamplesAsync(dtClient);
-
-            // Clean up
-
-            httpClient.Dispose();
         }
 
         /// <summary>
-        /// Illustrates how to construct a <see cref="DigitalTwinsClient"/>, using the <see cref="ClientSecretCredential"/>
+        /// Illustrates how to construct a <see cref="DigitalTwinsClient"/>, using the <see cref="DefaultAzureCredential"/>
         /// implementation of <see cref="Azure.Core.TokenCredential"/>.
-        /// </summary>
-        /// <param name="tenantId">The Id of the tenant of the application Id.</param>
-        /// <param name="clientId">The application Id.</param>
-        /// <param name="clientSecret">A client secret for the application Id.</param>
+        /// </summary>        
         /// <param name="adtEndpoint">The endpoint of the digital twins instance.</param>
         private static DigitalTwinsClient GetDigitalTwinsClient(string tenantId, string clientId, string clientSecret, string adtEndpoint)
         {
-            #region Snippet:DigitalTwinsSampleCreateServiceClientWithClientSecret
+            // These environment variables are necessary for DefaultAzureCredential to use application Id and client secret to login.
+            Environment.SetEnvironmentVariable("AZURE_CLIENT_SECRET", clientSecret);
+            Environment.SetEnvironmentVariable("AZURE_CLIENT_ID", clientId);
+            Environment.SetEnvironmentVariable("AZURE_TENANT_ID", tenantId);
 
-            // By using the ClientSecretCredential, a specified application Id can login using a
-            // client secret.
-            var tokenCredential = new ClientSecretCredential(
-                tenantId,
-                clientId,
-                clientSecret,
-                new TokenCredentialOptions { AuthorityHost = KnownAuthorityHosts.AzureCloud });
+            #region Snippet:DigitalTwinsSampleCreateServiceClientWithClientSecret
 
+            // DefaultAzureCredential supports different authentication mechanisms and determines the appropriate credential type based of the environment it is executing in.
+            // It attempts to use multiple credential types in an order till it finds a working credential.
+            var tokenCredential = new DefaultAzureCredential();
+            
             var client = new DigitalTwinsClient(
                 new Uri(adtEndpoint),
                 tokenCredential);
 
             #endregion Snippet:DigitalTwinsSampleCreateServiceClientWithClientSecret
 
             return client;
-        }
-
-        /// <summary>
-        /// Illustrates how to construct a <see cref="DigitalTwinsClient"/> including client options,
```

Comment:
I have removed this for now. If we really want to show different login mechanisms, we can do it later. It would use the DefaultAzureCredential but setup different values or environment variables. I have not been able to get it to work for interactive login yet.